components/actions_menu/actions_menu.tsx

```diff
@@ -0,0 +1,403 @@
+
+import React from 'react';
+import classNames from 'classnames';
+import {FormattedMessage, injectIntl, IntlShape} from 'react-intl';
+import './actions_menu.scss';
+
+import {Tooltip} from 'react-bootstrap';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
+
+import {Post} from 'mattermost-redux/types/posts';
+import {AppBinding} from 'mattermost-redux/types/apps';
+import {AppCallResponseTypes, AppCallTypes, AppExpandLevels} from 'mattermost-redux/constants/apps';
+
+import {DoAppCall, PostEphemeralCallResponseForPost} from 'types/apps';
+import {Locations, Constants, ModalIdentifiers} from 'utils/constants';
+import Permissions from 'mattermost-redux/constants/permissions';
+import ActionsTutorialTip from 'components/actions_menu/actions_menu_tutorial_tip';
+import MarketplaceModal from 'components/plugin_marketplace';
+import OverlayTrigger from 'components/overlay_trigger';
+import * as PostUtils from 'utils/post_utils';
+import * as Utils from 'utils/utils.jsx';
+import SystemPermissionGate from 'components/permissions_gates/system_permission_gate';
+import Pluggable from 'plugins/pluggable';
+import Menu from 'components/widgets/menu/menu';
+import MenuWrapper from 'components/widgets/menu/menu_wrapper';
+import {PluginComponent} from 'types/store/plugins';
+import {createCallContext, createCallRequest} from 'utils/apps';
+
+const MENU_BOTTOM_MARGIN = 80;
+
+export const PLUGGABLE_COMPONENT = 'PostDropdownMenuItem';
+type Props = {
+    appBindings: AppBinding[] | null;
+    appsEnabled: boolean;
+    currentTeamId: string;
+    handleDropdownOpened?: (open: boolean) => void;
+    intl: IntlShape;
+    isMenuOpen?: boolean;
+    isSysAdmin: boolean;
+    location?: 'CENTER' | 'RHS_ROOT' | 'RHS_COMMENT' | 'SEARCH' | string;
+    pluginMenuItems?: PluginComponent[];
+    post: Post;
+    showTutorialTip: boolean;
+    teamId?: string;
+    userId: string;
+
+    /**
+     * Components for overriding provided by plugins
+     */
+    components: {
+        [componentName: string]: PluginComponent[];
+    };
+
+    actions: {
+
+        /**
+         * Function to perform an app call
+         */
+        doAppCall: DoAppCall;
+
+        /**
+         * Function to open a modal
+         */
+        openModal: (modalData: {modalId: string; dialogType: any; dialogProps?: {post: Post; isRHS: boolean}}) => Promise<{
+            data: boolean;
+        }>;
+
+        /**
+         * Function to post the ephemeral message for a call response
+         */
+        postEphemeralCallResponseForPost: PostEphemeralCallResponseForPost;
+
+        /**
+         * Function to get the post menu bindings for this post.
+         */
+        fetchBindings: (userId: string, channelId: string, teamId: string) => Promise<{data: AppBinding[]}>;
+
+    }; // TechDebt: Made non-mandatory while converting to typescript
+}
+
+type State = {
+    openUp: boolean;
+    appBindings?: AppBinding[];
```

Comment:
Let me chime in for this one.

The idea is that app bindings are available in redux only for the RHS and the central channel. Among searches, each post may have its own set of bindings (since they belong to different channels). Therefore, if there are no bindings in the props, it will fetch them and store them in the state.